keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1543 @@
+        block_type: Must be one of `{"X", "Y", "Z"}`. For more details see the
+            papers "Designing network design spaces" and "Fast and Accurate
+            Model Scaling"
+        default_size: Default input image size.
+        model_name: An optional name for the model.
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(1/255.0)` layer.
+        include_top: Whether to include the fully-connected
+            layer at the top of the network.
+        classes: Optional number of classes to classify images
+            into, only to be specified if `include_top` is True, and
+            if no `weights` argument is specified.
+        weights: One of `None` (random initialization), or the path to the
+            weights file to be loaded. Defaults to `None`.
+        input_tensor: Optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+        input_shape: Optional shape tuple, defaults to (None, None, 3).
+            It should have exactly 3 inputs channels.
+        pooling: Optional pooling mode for feature extraction
+            when `include_top` is `False`. Defaults to None.
+            - `None` means that the output of the model will be
+                the 4D tensor output of the
+                last convolutional layer.
+            - `avg` means that global average pooling
+                will be applied to the output of the
+                last convolutional layer, and thus
+                the output of the model will be a 2D tensor.
+            - `max` means that global max pooling will
+                be applied.
+        classifier_activation: A `str` or callable. The activation function to
+            use on the "top" layer. Ignored unless `include_top=True`. Set
+            `classifier_activation=None` to return the logits of the "top"
+            layer. Defaults to `"softmax"`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights is None or tf.io.gfile.exists(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` (random initialization) "
+            "or the path to the weights file to be loaded."
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `classes`. "
+            f"Received: classes={classes}"
+        )
+
+    if include_top and pooling:
+        raise ValueError(
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    img_input = utils.parse_model_inputs(input_shape, input_tensor)
+    x = img_input
+
+    if include_rescaling:
+        x = layers.Rescaling(scale=1.0 / 255.0)(x)
+    x = Stem(name=model_name)(x)
+
+    in_channels = 32  # Output from Stem
```

Comment:
Can't it be programmatically calculated?

Reply:
`x.shape[-1]`?